test/functional/tests/io_class/test_io_class_service_support.py

```diff
@@ -0,0 +1,285 @@
+        saved_usage_stats = cache.get_io_class_statistics(io_class_id=0, stat_filter=[
+            StatsFilter.usage]).usage_stats
+        saved_conf_stats = cache.get_io_class_statistics(io_class_id=0, stat_filter=[
+            StatsFilter.conf]).config_stats
+
+    with TestRun.step("Create init config from running CAS configuration."):
+        InitConfig.create_init_config_from_running_configuration(
+            cache_extra_flags=f"ioclass_file={ioclass_config_path}")
+        os_utils.sync()
+
+    with TestRun.step(f"Reboot system to runlevel {runlevel}."):
+        os_utils.change_runlevel(runlevel)
+        TestRun.executor.reboot()
+
+    with TestRun.step("Check if CAS device loads properly - "
+                      "IO class configuration and statistics shall not change"):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 1:
+            TestRun.fail("Cache did not start at boot time.")
+        cache = caches[0]
+        cores = casadm_parser.get_cores(cache.cache_id)
+        if len(cores) != 1:
+            TestRun.fail(f"Actual number of cores: {len(cores)}\nExpected number of cores: 1")
+        core = cores[0]
+        output_io_classes = cache.list_io_classes()
+        compare_io_classes_list(config_io_classes, output_io_classes)
+
+        # Reads from core can invalidate some data so it is possible that occupancy after reboot
+        # is lower than before
+        reads_from_core = cache.get_statistics(stat_filter=[StatsFilter.blk]).block_stats.core.reads
+        read_usage_stats = cache.get_io_class_statistics(io_class_id=0, stat_filter=[
+            StatsFilter.usage]).usage_stats
+        read_conf_stats = cache.get_io_class_statistics(io_class_id=0, stat_filter=[
+            StatsFilter.conf]).config_stats
+
+        if read_conf_stats != saved_conf_stats:
+            TestRun.LOGGER.error(f"Statistics do not match. Before: {str(saved_conf_stats)} "
+                                 f"After: {str(read_conf_stats)}")
+        if read_usage_stats != saved_usage_stats and \
+                saved_usage_stats.occupancy - read_usage_stats.occupancy > reads_from_core:
+            TestRun.LOGGER.error(f"Statistics do not match. Before: {str(saved_usage_stats)} "
+                                 f"After: {str(read_usage_stats)}")
+
+    with TestRun.step("Mount CAS device and run IO again."):
+        core.mount(mountpoint)
+        run_io()
+
+    with TestRun.step("Check that data are mostly read from cache."):
+        cache_stats = cache.get_statistics()
+        read_hits = cache_stats.request_stats.read.hits
+        read_total = cache_stats.request_stats.read.total
+        read_hits_percentage = read_hits / read_total * 100
+        if read_hits_percentage <= 50:
+            TestRun.LOGGER.error(f"Read hits percentage too low: {read_hits_percentage}%\n"
+                                 f"Read hits: {read_hits}, read total: {read_total}")
```

Comment:
As I removed iterating over cache modes and the test below, I think I will left here checking those statistics, because to separate this into the other test I should copy almost all of this test and check those stats at the end, what does not make much sense for me :D This test was meant to check statistics after loading CAS with use of service, so I think it is good place for that